sbi/utils/restriction_estimator.py

```diff
@@ -504,57 +512,183 @@ def _converged(self, epoch: int, stop_after_epochs: int) -> bool:
         return converged
 
 
+def threshold_distribution(
+    dist: Any,
+    quantile: float = 1e-4,
+    num_samples_to_estimate_support: int = 1_000_000,
+) -> Callable:
+    """Return function that thresholds a density at a particular `1-quantile`.
+
+    Args:
+        density: Probability distribution to be thresholded, must have `.sample()` and
+            `.log_prob()`.
+        quantile: The returned will be `True` within the `1-quantile`
+            high-probability region of the density. In other words: `quantile` is the
+            fraction of mass that is excluded from `dist`.
+        num_samples_to_estimate_support: The number of samples that are drawn from
+            `dist` in order to obtain the threshold. Higher values are more accurate
+            but slower.
+
+    Returns:
+        Callabe which is true for $\theta$ in the `1-quantile` high-probability region
+        of the `dist`.
+    """
+    samples = dist.sample((num_samples_to_estimate_support,))
+    log_probs = dist.log_prob(samples)
+    sorted_log_probs, _ = torch.sort(log_probs)
+    log_prob_threshold = sorted_log_probs[
+        int(quantile * num_samples_to_estimate_support)
+    ]
+
+    def accept_reject(theta: Tensor) -> Tensor:
+        theta_log_probs = dist.log_prob(theta)
+        predictions = theta_log_probs > log_prob_threshold
+        return predictions.bool()
+
+    return accept_reject
+
+
+def threshold_classifier(
+    classifier: Any,
+    validation_theta: Tensor,
+    validation_label: Tensor,
+    allowed_false_negatives: Optional[float] = None,
+    reweigh_factor: Optional[float] = None,
+    print_fp_rate=False,
+    safety_margin: Optional[Union[str, float]] = "frequentist",
+) -> Callable:
+    r"""
+    Set the decision threshold of the classifier.
+
+    Compute the highest decision threshold at which the number of false negatives
+    is smaller than `allowed_false_negatives`. This threshold will then be set as
+    default when calling `.forward()` and for `.build_sim_informed_prior()`.
+
+    Args:
+        allowed_false_negatives: Allowed fraction of false negatives on a held-out
+            test set.
+        reweigh_factor: Post-hoc correction factor. Should be in [0, 1]. A large
+            reweigh factor will increase the probability of predicting a `invalid`
+            simulation.
+        print_fp_rate: Whether or not to compute and print the false-positive rate
```

Comment:
put this into a `verbose` arg?

Reply:
since this is the only thing that is printed or shown here I think it is fine to keep it specific to the `fp_rate`